src/explorer/appServiceNode.ts

```diff
@@ -85,6 +85,34 @@ export class AppServiceNode extends NodeBase {
         return this.start();
     }
 
+    async delete(azureAccount: AzureAccountWrapper): Promise<boolean> {
+        let servicePlanName = this.site.serverFarmId.substring(this.site.serverFarmId.lastIndexOf('/serverfarms/') + '/serverfarms/'.length);
+        let servicePlanRG = this.site.serverFarmId.substring(this.site.serverFarmId.indexOf('resourceGroups/') + 'resourceGroups/'.length, this.site.serverFarmId.indexOf('/providers/'));
```

Comment:
It's unlikely, but this doesn't work for a resource group named 'providers'. What if you used a regex isntead:
```
const matches = this.site.serverFarmId.match("^\/subscriptions\/.*\/resourceGroups\/(.*)\/providers\/Microsoft\.Web\/serverfarms\/(.*)");
const servicePlanRG = matches[1];
const servicePlanName = matches[2];
```